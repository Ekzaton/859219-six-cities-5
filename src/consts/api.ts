export const BASE_URL = `https://6.react.pages.academy/six-cities`;
export const REQUEST_TIMEOUT = 5000;
