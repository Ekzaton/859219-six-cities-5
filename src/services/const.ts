export const BASE_URL = `https://5.react.pages.academy/six-cities`;
export const REQUEST_TIMEOUT = 5000;

export enum HTTPCode {
  BAD_REQUEST = 400,
  UNAUTHORIZED = 401,
  NOT_FOUND = 404
}
