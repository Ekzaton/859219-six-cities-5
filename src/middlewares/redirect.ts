import {Middleware} from "redux";

import {ActionType} from "../consts/store";

import history from "../history/history";

export const redirect: Middleware = () => (dispatch) => (action) => {
  if (action.type === ActionType.SET_REDIRECT_TO_ROUTE) {
    history.push(action.payload);
  }

  return dispatch(action);
};
