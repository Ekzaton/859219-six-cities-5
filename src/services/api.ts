import axios, {AxiosError, AxiosInstance, AxiosResponse} from "axios";
import camelcaseKeys from "camelcase-keys";

import {BASE_URL, REQUEST_TIMEOUT} from "../consts/api";

export const createAPI = (): AxiosInstance => {
  const api = axios.create({
    baseURL: BASE_URL,
    timeout: REQUEST_TIMEOUT,
    withCredentials: true,
  });

  const onSuccess = (response: AxiosResponse) => {
    return {...response, data: camelcaseKeys(response.data, {deep: true})};
  };

  const onFail = (error: AxiosError) => {
    throw error;
  };

  api.interceptors.response.use(onSuccess, onFail);

  return api;
};
