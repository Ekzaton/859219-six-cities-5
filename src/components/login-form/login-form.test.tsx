import React from 'react';
import {Provider} from 'react-redux';
import createMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {render} from '@testing-library/react';

import {redirect} from "../../middlewares/redirect";
import {createAPI} from "../../services/api";

import LoginForm from "./login-form";

describe(`Login form renders correctly`, () => {
  const api = createAPI();
  const thunkWithAPI = thunk.withExtraArgument(api);
  const mockStore = createMockStore([thunkWithAPI, redirect]);

  it(`for not authorized user`, () => {
    const store = mockStore({
      login: {
        isSending: false,
        sendingError: null
      }
    });

    const {container} = render(
        <Provider store={store}>
          <LoginForm/>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it(`for not authorized user (sending)`, () => {
    const store = mockStore({
      login: {
        isSending: true,
        sendingError: null
      }
    });

    const {container} = render(
        <Provider store={store}>
          <LoginForm/>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it(`for not authorized user (sending error)`, () => {
    const error = {
      data: {},
      status: 400,
      statusText: `Bad Request`,
      headers: [],
      config: {}
    };

    const store = mockStore({
      login: {
        isSending: true,
        sendingError: error
      }
    });

    const {container} = render(
        <Provider store={store}>
          <LoginForm/>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
