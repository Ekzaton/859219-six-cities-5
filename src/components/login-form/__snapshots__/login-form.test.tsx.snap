// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login form renders correctly for not authorized user (sending error) 1`] = `
<div>
  <form
    class="login__form form"
  >
    <div
      class="login__input-wrapper form__input-wrapper"
    >
      <label
        class="visually-hidden"
      >
        E-mail
      </label>
      <input
        autocomplete="false"
        class="login__input form__input"
        disabled=""
        name="email"
        placeholder="Email"
        required=""
        type="email"
      />
    </div>
    <div
      class="login__input-wrapper form__input-wrapper"
    >
      <label
        class="visually-hidden"
      >
        Password
      </label>
      <input
        autocomplete="false"
        class="login__input form__input"
        disabled=""
        name="password"
        placeholder="Password"
        required=""
        type="password"
      />
    </div>
    <button
      class="login__submit form__submit button"
      disabled=""
      type="submit"
    >
      Sign in
    </button>
    <div
      style="margin-top: 30px;"
    >
      Please enter your email and password
    </div>
    <div
      style="margin-top: 30px;"
    />
  </form>
</div>
`;

exports[`Login form renders correctly for not authorized user (sending) 1`] = `
<div>
  <form
    class="login__form form"
  >
    <div
      class="login__input-wrapper form__input-wrapper"
    >
      <label
        class="visually-hidden"
      >
        E-mail
      </label>
      <input
        autocomplete="false"
        class="login__input form__input"
        disabled=""
        name="email"
        placeholder="Email"
        required=""
        type="email"
      />
    </div>
    <div
      class="login__input-wrapper form__input-wrapper"
    >
      <label
        class="visually-hidden"
      >
        Password
      </label>
      <input
        autocomplete="false"
        class="login__input form__input"
        disabled=""
        name="password"
        placeholder="Password"
        required=""
        type="password"
      />
    </div>
    <button
      class="login__submit form__submit button"
      disabled=""
      type="submit"
    >
      Sign in
    </button>
    <div
      style="margin-top: 30px;"
    >
      Please enter your email and password
    </div>
    <div
      style="margin-top: 30px;"
    />
  </form>
</div>
`;

exports[`Login form renders correctly for not authorized user 1`] = `
<div>
  <form
    class="login__form form"
  >
    <div
      class="login__input-wrapper form__input-wrapper"
    >
      <label
        class="visually-hidden"
      >
        E-mail
      </label>
      <input
        autocomplete="false"
        class="login__input form__input"
        name="email"
        placeholder="Email"
        required=""
        type="email"
      />
    </div>
    <div
      class="login__input-wrapper form__input-wrapper"
    >
      <label
        class="visually-hidden"
      >
        Password
      </label>
      <input
        autocomplete="false"
        class="login__input form__input"
        name="password"
        placeholder="Password"
        required=""
        type="password"
      />
    </div>
    <button
      class="login__submit form__submit button"
      disabled=""
      type="submit"
    >
      Sign in
    </button>
    <div
      style="margin-top: 30px;"
    >
      Please enter your email and password
    </div>
    <div
      style="margin-top: 30px;"
    />
  </form>
</div>
`;
